src/VisualStudio/Core/Def/Implementation/Remote/RemoteHost.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Execution;
+using Microsoft.CodeAnalysis.Shared.Options;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.Remote
+{
+    internal abstract class RemoteHost
+    {
+        private readonly Workspace _workspace;
+
+        public RemoteHost(Workspace workspace)
+        {
+            _workspace = workspace;
+        }
+
+        public event EventHandler<bool> ConnectionChanged;
+
+        public async Task<Session> CreateSnapshotSessionAsync(Solution solution, CancellationToken cancellationToken)
```

Comment:
This should probably return a ```Task<SnapshotSessions>``` or be called ```CreateSessionAsync```.  

Note: "StartSessionAsync" also sounds good.